File: perf/bench/primes/primes.go
package main

import "fmt"

func isPrime(n int64) bool {
	if n < 2 {
		return false
	}
	for d := int64(2); d*d <= n; d++ {
		if n%d == 0 {
			return false
		}
	}
	return true
}

func main() {
	var count int64 = 0
	for n := int64(2); n < 3000000; n++ {
		if isPrime(n) {
			count++
		}
	}
	fmt.Println(count)
}

File: perf/bench/primes/primes.swift
func isPrime(_ n: Int64) -> Bool {
    if n < 2 { return false }
    var d: Int64 = 2
    while d * d <= n {
        if n % d == 0 { return false }
        d += 1
    }
    return true
}
var count: Int64 = 0
var n: Int64 = 2
while n < 3000000 {
    if isPrime(n) { count += 1 }
    n += 1
}
print(count)
